hbase-client/src/main/java/org/apache/hadoop/hbase/exceptions/ClientExceptionsUtil.java

```diff
@@ -59,8 +59,7 @@ public static boolean isMetaClearingException(Throwable cur) {
     if (cur == null) {
       return true;
     }
-    return !isSpecialException(cur) || (cur instanceof RegionMovedException)
-        || cur instanceof NotServingRegionException;
+    return !regionDefinitelyOnTheRegionServerException(cur);
   }
 
   public static boolean isSpecialException(Throwable cur) {
```

Comment:
Make this private? And can we reuse the regionDefinitelyOnTheRegionServerException so we can write less instance of here? And MultiActionResultTooLarge is not included in regionDefinitelyOnTheRegionServerException? Why?

Reply:
OK, I misunstood your comment, and thought your concern is the removal of  the public method isSpecialException.  1) The exceptions make isSpecialException true are  a subset of  the complementary set of exceptions that make regionDefinitelyOnTheRegionServerException true. So will use !regionDefinitelyOnTheRegionServerException  instead of isSpecialException.
2) MultiActionResultTooLarge is a subclass of RetryImmediatelyException, which is there.

Reply:
MultiActionResultTooLarge  is a subclass of  RetryImmediatelyException, which is there.

Reply:
> Make this private? And can we reuse the regionDefinitelyOnTheRegionServerException so we can write less instance of here? And MultiActionResultTooLarge is not included in regionDefinitelyOnTheRegionServerException? 

Here the isSpecialException is equivalent with regionDefinitelyOnTheRegionServerException, the two more exceptions in isSpecialException  are RegionMovedException and  RegionOpeningException. Both of them are subclasses of NotServingRegionException, which is there.

Reply:
OK, the problem for using regionDefinitelyOnTheRegionServerException directly in findException is the name of the method... Maybe we could add a other method which is called something like 'the exception we care', as the intention there is that, if we hit these exceptions then we will not trying to get the cause any more.